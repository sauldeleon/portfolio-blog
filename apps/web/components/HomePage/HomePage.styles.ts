import styled from 'styled-components'

import { CircleLink } from '@sdlgr/circle-link'
import { Heading } from '@sdlgr/typography'

export const StyledHeading = styled(Heading)`
  margin-bottom: 30px;
  text-align: center;

  ${({ theme }) => theme.media.up.md} {
    ${({ theme }) => theme.typography.heading.heading1}
  }

  ${({ theme }) => theme.media.up.lg} {
    margin-bottom: 50px;
  }
`

export const StyledSubHeading = styled(Heading)`
  margin-top: 50px;
  margin-bottom: 50px;
`

export const StyledCircleLink = styled(CircleLink)`
  margin-bottom: 30px;

  ${({ theme }) => theme.media.up.md} {
    margin-bottom: 50px;
  }
`
