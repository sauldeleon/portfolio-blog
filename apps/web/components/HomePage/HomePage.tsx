import { SLLogo } from '@sdlgr/assets'

import { useTranslation } from '@web/i18n/client'

import { StyledHeading, StyledPage } from './HomePage.styles'

export function HomePage() {
  const { t } = useTranslation()
  return (
    <StyledPage>
      <StyledHeading>{t('underConstruction')}</StyledHeading>
      <SLLogo />
    </StyledPage>
  )
}
