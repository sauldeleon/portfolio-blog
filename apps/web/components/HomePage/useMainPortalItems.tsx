import { useMemo } from 'react'

import {
  Auth0Icon,
  CSS3Icon,
  ChromeIcon,
  CypressIcon,
  DockerIcon,
  EslintIcon,
  GitIcon,
  GithubIcon,
  GitlabIcon,
  HTML5Icon,
  JavascriptIcon,
  JestIcon,
  JsonIcon,
  MaterialUiIcon,
  MongoDBIcon,
  NextJSIcon,
  NodeJSIcon,
  NpmIcon,
  NxIcon,
  PostgreSqlIcon,
  PrettierIcon,
  ReactJSIcon,
  ReactQueryIcon,
  SentryIcon,
  SpiderManIcon,
  StorybookIcon,
  StyledComponentsIcon,
  TypescriptIcon,
  VSCodeIcon,
  VercelIcon,
  WebpackIcon,
  XataIOIcon,
  YarnIcon,
} from '@sdlgr/assets'

import { AnimatedItemProps } from '@web/components/AnimatedItem/AnimatedItem'

export function useMainPortalItems() {
  const items: AnimatedItemProps[] = useMemo(
    () => [
      {
        svg: <NodeJSIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <CSS3Icon />,
        rotate: true,
        colorSwap: true,
        size: 'S',
      },
      {
        svg: <GithubIcon />,
        rotate: true,
        colorSwap: true,
        size: 'S',
      },
      {
        svg: <TypescriptIcon />,
        rotate: true,
        colorSwap: true,
        size: 'S',
      },
      {
        svg: <NextJSIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <ReactJSIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <MongoDBIcon />,
        rotate: true,
        colorSwap: true,
        size: 'S',
      },
      {
        svg: <ReactQueryIcon />,
        rotate: true,
      },
      {
        svg: <JestIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <NxIcon />,
        rotate: true,
        size: 'S',
        colorSwap: true,
      },
      {
        svg: <YarnIcon />,
        rotate: true,
        size: 'S',
        colorSwap: true,
      },
      {
        svg: <XataIOIcon />,
        rotate: true,
        size: 'S',
      },
      {
        svg: <StyledComponentsIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <EslintIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <CypressIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <PrettierIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <DockerIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <HTML5Icon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <JavascriptIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <NpmIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <GitIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <ChromeIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <VSCodeIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <MaterialUiIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <PostgreSqlIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <SentryIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <StorybookIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <WebpackIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <GitlabIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <JsonIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <VercelIcon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <Auth0Icon />,
        rotate: true,
        colorSwap: true,
      },
      {
        svg: <SpiderManIcon />,
        rotate: true,
        colorSwap: true,
      },
    ],
    []
  )

  return items
}
