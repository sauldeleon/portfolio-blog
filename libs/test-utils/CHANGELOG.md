# Changelog Test utils

## [0.2.3](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.2...test-utils-0.2.3) (2024-01-08)

## [0.2.2](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.1...test-utils-0.2.2) (2023-09-26)

## [0.2.1](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.2.0...test-utils-0.2.1) (2023-09-14)

# [0.2.0](https://github.com/sauldeleon/portfolio-blog/compare/test-utils-0.1.0...test-utils-0.2.0) (2023-09-05)

### Bug Fixes

- reduced duplicated lines ([f51168a](https://github.com/sauldeleon/portfolio-blog/commit/f51168aefe025a5a65ffdb2dc22f7a16b863406e))
- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))

# 0.1.0 (2023-09-02)

### Bug Fixes

- rename commit message format ([e1e2ab4](https://github.com/sauldeleon/portfolio-blog/commit/e1e2ab404bbd2c32f3508d1ed8197b3fbff93cb9))

### Features

- add contact initial page ([fea862b](https://github.com/sauldeleon/portfolio-blog/commit/fea862b3507268013149050ae3f05fa517e7b3d9))
- add homepage ([60523aa](https://github.com/sauldeleon/portfolio-blog/commit/60523aa31ec088d318745ef0c757b4cbe656f0d4))
- deleted changelogs ([e9be338](https://github.com/sauldeleon/portfolio-blog/commit/e9be33836ee47b6505ad94d21f4be21855a7fa0d))
- improved a11y and styles ([873f656](https://github.com/sauldeleon/portfolio-blog/commit/873f656587f50df8e29ad526fdc68fea0b2c9784))
- toggle language button ([afd97e9](https://github.com/sauldeleon/portfolio-blog/commit/afd97e95beac0f70ed643c753e71db2ca014054b))
