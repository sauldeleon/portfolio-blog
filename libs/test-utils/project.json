{
  "name": "test-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/test-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "version": {
      "executor": "@jscutlery/semver:version",
      "options": {
        "changelogHeader": "# Changelog Test utils \n",
        "baseBranch": "main",
        "commitMessageFormat": "chore(${projectName}): release: v${version}",
        "push": false
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/test-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/test-utils/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
