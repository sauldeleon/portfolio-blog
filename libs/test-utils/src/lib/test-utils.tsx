import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { Matcher } from '@testing-library/dom/types/matches'
import {
  RenderOptions,
  RenderResult,
  queryAllByAttribute as _queryAllByAttribute,
  queryByAttribute as _queryByAttribute,
  act,
  render,
  screen,
} from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'
import Router, { NextRouter } from 'next/router'
import { ReactNode, useEffect, useState } from 'react'
import { ThemeProvider } from 'styled-components'

import { LanguageContextProvider } from '@sdlgr/i18n-config'
import { mainTheme } from '@sdlgr/main-theme'

export type MatcherCustomFnParams = {
  content: string
  element: Element | null
}

/* istanbul ignore next */
export const renderWithTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={mainTheme}>{children}</ThemeProvider>)
}

/* istanbul ignore next */
export function RenderProviders({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <LanguageContextProvider value={{ language: 'en' }}>
        <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
      </LanguageContextProvider>
    </QueryClientProvider>
  )
}

/* istanbul ignore next */
export const renderApp = (
  children: ReactNode,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(<RenderProviders>{children}</RenderProviders>, options)
}

/**
 * Generic function to get element by attribute and value
 * @example queryByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function queryByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example queryByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function queryAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Generic function to get element by attribute and value
 * @example getByAttribute('id')(baseElement, 'element-id')
 * @param {string} attr
 * @return {any}
 */
export function getByAttribute(attr: string) {
  return _queryByAttribute.bind(null, attr)
}

/**
 * Generic function to get all elements by attribute and value
 * @example getByAttribute('class')(baseElement, 'element-class')
 * @param {string} attr
 * @returns {any}
 */
export function getAllByAttribute(attr: string) {
  return _queryAllByAttribute.bind(null, attr)
}

/**
 * Shortcut version for queryByAttribute, but specific for id
 * @param {HTMLElement} container
 * @param {string} id
 * @return {any}
 */
export function getById(container: HTMLElement, id: string) {
  return queryByAttribute('id')(container, id) as Element
}

/**
 * Shortcut version for queryByAttribute, but specific for class name
 * @param {HTMLElement} container
 * @param {string} className
 * @return {any}
 */
export function getByClassName(container: HTMLElement, className: string) {
  return queryByAttribute('class')(container, className) as Element
}

function matcherCustomFn(
  tag: string,
  matcher: Matcher,
  { element, content }: MatcherCustomFnParams
) {
  const isTag = element?.tagName.toUpperCase() === tag.toUpperCase()
  let matching = false
  if (isTag) {
    matching = content === matcher
    /* istanbul ignore else*/
    if (typeof matcher === 'function') {
      matching = matcher(content, element)
    } else if (matcher instanceof RegExp) {
      matching = matcher.test(content)
    }
  }
  return matching && isTag
}

/**
 * Get 1 element by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getByTagAndText(tag: string, matcher: Matcher) {
  return screen.getByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Get multiple elements by tag and text
 * @param {string} tag
 * @param {Matcher} matcher
 * @return {any}
 */
export function getAllByTagAndText(tag: string, matcher: Matcher) {
  return screen.getAllByText((content, element) =>
    matcherCustomFn(tag, matcher, { content, element })
  )
}

/**
 * Mock Math.random() to return a specific value
 * @param {number} value
 */
export function mockRandom(value = 0.1234) {
  const mockMath = Object.create(global.Math)
  mockMath.random = () => value
  global.Math = mockMath
}

/**
 * Hook to use in test to ensure "last" render is called
 */
export function useWithRefreshInTest() {
  const [withRefreshInTest, setWithRefreshInTest] = useState(false)

  useEffect(() => {
    // Change state to "rerender" the component
    setWithRefreshInTest(true)
  }, [])

  return withRefreshInTest
}

export { render } from '@testing-library/react'

/* istanbul ignore next */
export const awaitAllAsync = async () =>
  act(() => new Promise((resolve) => setTimeout(resolve, 0)))

/* istanbul ignore next */
export function renderWithNextRouter(
  children: ReactNode,
  options?: Partial<NextRouter>
) {
  return renderWithTheme(
    <RouterContext.Provider value={{ ...Router, ...options }}>
      {children}
    </RouterContext.Provider>
  )
}
