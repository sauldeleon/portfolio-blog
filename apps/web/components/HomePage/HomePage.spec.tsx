import { screen } from '@testing-library/react'

import { renderApp } from '@sdlgr/test-utils'

import { HomePage } from './HomePage'

describe('HomePage', () => {
  it('should render successfully', async () => {
    renderApp(<HomePage />)
    const items = await screen.findAllByRole('presentation')
    expect(items.length).toBe(2)
  })
})
