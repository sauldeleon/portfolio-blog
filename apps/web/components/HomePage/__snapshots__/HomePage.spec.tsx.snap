// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render successfully 1`] = `
.c17 {
  font-family: 'Roboto mono',monospace;
  font-weight: 400;
  color: #FFFFFF;
  font-size: 3rem;
  line-height: normal;
  margin: 0;
}

.c22 {
  font-family: 'Roboto mono',monospace;
  font-weight: 400;
  color: #FFFFFF;
  font-size: 1.5rem;
  line-height: normal;
  margin: 0;
}

.c8 {
  font-family: 'Roboto mono',monospace;
  font-weight: 400;
  color: #FFFFFF;
  font-size: 0.9rem;
  line-height: normal;
  white-space: pre-line;
}

.c8 ul,
.c8 ol {
  margin-left: 40px;
  margin-bottom: 10px;
}

.c8 ul {
  list-style: outside disc;
}

.c8 ol {
  list-style: decimal;
}

.c8 li {
  padding-left: 10px;
}

.c21 {
  font-family: 'Roboto mono',monospace;
  font-weight: 500;
  color: #FFFFFF;
  font-size: 0.5rem;
  line-height: normal;
  text-transform: uppercase;
}

.c19 {
  outline: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  max-height: 76px;
}

.c19:hover circle.loading {
  stroke: white;
  stroke-width: 6px;
  stroke-dasharray: 110;
  stroke-dashoffset: 82;
  animation: clock-loading 0.5s steps(4, jump-none) forwards;
  transform: rotate(-90deg);
  transform-origin: center;
}

.c10 {
  outline: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  max-height: 42px;
}

.c10:hover circle.loading {
  stroke: white;
  stroke-width: 6px;
  stroke-dasharray: 110;
  stroke-dashoffset: 82;
  animation: clock-loading 0.5s steps(4, jump-none) forwards;
  transform: rotate(-90deg);
  transform-origin: center;
}

.c20 {
  position: relative;
  width: 76px;
  height: 76px;
}

.c12 {
  position: relative;
  width: 42px;
  height: 42px;
}

.c13 {
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
}

.c14 {
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  display: flex;
}

.c15 {
  margin-left: 8px;
}

.c1 {
  background-color: #FFFFFF;
  color: #000000;
  height: 130px;
  max-width: 1440px;
  width: 100%;
  padding: 0 15px;
  margin: 0 auto;
  display: flex;
  justify-items: center;
  align-items: end;
}

.c2 {
  display: flex;
}

.c6 {
  display: flex;
}

.c3 {
  color: #000000;
  margin-bottom: 25px;
}

.c4 {
  display: flex;
  align-items: end;
  flex-grow: 1;
  justify-content: center;
  height: 100%;
}

.c5 {
  width: 59px;
  display: flex;
  justify-content: center;
  height: 100%;
  padding-bottom: 25px;
}

.c5 .c7 {
  font-family: 'Roboto mono',monospace;
  font-weight: 500;
  color: #FFFFFF;
  font-size: 0.55rem;
  line-height: normal;
  color: #000000;
  writing-mode: tb-rl;
  transform: rotate(-180deg);
}

.c5:hover {
  background-color: #000000;
  cursor: pointer;
}

.c5:hover .c7 {
  color: #FFFFFF;
}

.c9 {
  width: 59px;
  display: flex;
  justify-content: center;
  height: 100%;
  padding-bottom: 25px;
}

.c9 .c7 {
  font-family: 'Roboto mono',monospace;
  font-weight: 500;
  color: #FFFFFF;
  font-size: 0.55rem;
  line-height: normal;
  color: #000000;
  writing-mode: tb-rl;
  transform: rotate(-180deg);
}

.c9:hover {
  background-color: #000000;
  cursor: pointer;
}

.c9:hover .c7 {
  color: #FFFFFF;
}

.c11 {
  display: none;
}

.c0 {
  width: 100%;
  display: flex;
  flex-direction: column;
}

.c16 {
  display: flex;
  min-height: calc(100vh - 115px);
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.c18 {
  margin-bottom: 50px;
}

@media (min-width: 768px) {
  .c1 {
    align-items: center;
    background-color: transparent;
    color: #FFFFFF;
    height: 115px;
  }
}

@media (min-width: 1024px) {
  .c1 {
    padding: 0 30;
  }
}

@media (min-width: 1440px) {
  .c1 {
    padding: 0;
  }
}

@media (min-width: 768px) {
  .c6 {
    display: unset;
  }
}

@media (min-width: 768px) {
  .c3 {
    color: #FFFFFF;
    margin-bottom: 0;
  }
}

@media (min-width: 768px) {
  .c4 {
    gap: 30px;
    align-items: center;
    height: unset;
  }
}

@media (min-width: 768px) {
  .c5 {
    width: auto;
    padding-bottom: 0;
  }

  .c5 .c7 {
    font-family: 'Roboto mono',monospace;
    font-weight: 400;
    color: #FFFFFF;
    font-size: 0.9rem;
    line-height: normal;
    color: #FFFFFF;
    writing-mode: unset;
    transform: unset;
  }
}

@media (min-width: 768px) {
  .c9 {
    display: none;
    width: auto;
    padding-bottom: 0;
  }

  .c9 .c7 {
    font-family: 'Roboto mono',monospace;
    font-weight: 400;
    color: #FFFFFF;
    font-size: 0.9rem;
    line-height: normal;
    color: #FFFFFF;
    writing-mode: unset;
    transform: unset;
  }
}

@media (min-width: 768px) {
  .c11 {
    display: flex;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <nav
        class="c1"
      >
        <a
          class="c2"
          href="/"
        >
          <svg
            class="c3"
            height="55"
          >
            slLogo.svg
          </svg>
        </a>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <a
              class="c6"
              href="/experience"
            >
              <p
                class="c7 c8"
              >
                Experience
              </p>
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              class="c6"
              href="/contact"
            >
              <p
                class="c7 c8"
              >
                Contact
              </p>
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              class="c6"
              href="/portfolio"
            >
              <p
                class="c7 c8"
              >
                CV/Portfolio
              </p>
            </a>
          </li>
          <li
            class="c9"
          >
            <a
              class="c6"
              href="/contact"
            >
              <p
                class="c7 c8"
              >
                Get in touch
              </p>
            </a>
          </li>
        </ul>
        <a
          class="c10 c11"
          href="/contact"
        >
          <div
            class="c12"
          >
            <svg
              class="c13"
              height="42"
              width="42"
            >
              circleButton.svg
            </svg>
            <div
              class="c14"
            >
              <svg>
                arrowRight.svg
              </svg>
            </div>
          </div>
          <p
            class="c7 c8 c15"
          >
            Get in touch
          </p>
        </a>
      </nav>
      <div
        class="c16"
      >
        <h1
          class="c17 c18"
        >
          Saúl de León Guerrero
        </h1>
        <a
          class="c19"
          href="/"
        >
          <div
            class="c20"
          >
            <svg
              class="c13"
              height="76"
              width="76"
            >
              circleButton.svg
            </svg>
            <div
              class="c14"
            >
              <div
                class="c21"
              >
                Explore
              </div>
            </div>
          </div>
        </a>
        <h2
          class="c22 c18"
        >
          Under construction
        </h2>
        <svg>
          slLogo.svg
        </svg>
      </div>
    </div>
  </div>
</body>
`;
